src/battle-scene.ts

```diff
@@ -2151,12 +2152,23 @@ export default class BattleScene extends SceneBase {
   }
 
   addMoney(amount: integer): void {
-    this.money = Math.min(this.money + amount, Number.MAX_SAFE_INTEGER);
+    const mysteryIncomeAura = this.mysteryEncounterAuras.FindAura(AuraType.INCOME);
```

Comment:
Just leaving a comment here so I don’t forget, this functionality should probably be moved to an applyModifiers() function